onnxruntime/test/optimizer/qdq_test_utils.cc

```diff
@@ -0,0 +1,109 @@
+  auto* q_output = builder.MakeIntermediate();
+  auto* dq_output = builder.MakeIntermediate();
+  builder.AddQuantizeLinearNode(q_input, scale, q_output);
+  builder.AddDequantizeLinearNode<T>(q_output, scale, dq_output);
+  return dq_output;
+}
+
+// TODO: for now it just builds a conv qdq graph. 
+// can be modified and made it shared among different qdq test graphs associated with other operators
+template <typename InputType, typename WeightType, typename BiasType, typename OutputType>
+BuildTestCaseFn BuildTestCase(const std::vector<int64_t>& input_shape, const std::vector<int64_t>& weights_shape) {
+  return [&](ModelTestBuilder& builder) {
+    auto* input_arg = builder.MakeInput<float>(input_shape, -1.f, 1.f);
+    auto* output_arg = builder.MakeOutput();
+
+    typedef std::numeric_limits<InputType> InputLimits;
+    typedef std::numeric_limits<WeightType> WeightLimits;
+    typedef std::numeric_limits<OutputType> OutputLimits;
+
+    InputType input_min_value = InputLimits::min();
+    InputType input_max_value = InputLimits::max();
+
+    WeightType weight_min_value = WeightLimits::min();
+    WeightType weight_max_value = WeightLimits::max();
+    if (std::is_same<WeightType, int8_t>::value) {
+      weight_min_value /= 2;
+      weight_max_value /= 2;
+    }
+
+    auto* dq_w_output = builder.MakeIntermediate();
+    auto* weight = builder.MakeInitializer<WeightType>(weights_shape, weight_min_value, weight_max_value);
+    builder.AddDequantizeLinearNode<WeightType>(weight, .03f,
+                                                (weight_min_value + weight_max_value) / 2 + 1,
+                                                dq_w_output);
+
+    auto* dq_bias_output = builder.MakeIntermediate();
+    auto* bias = builder.MakeInitializer<BiasType>({weights_shape[0]}, static_cast<BiasType>(0), static_cast<BiasType>(127));
+    builder.AddDequantizeLinearNode<BiasType>(bias, .0012f,
+                                              0,
+                                              dq_bias_output);
+
+    auto* conv_output = builder.MakeIntermediate();
+    auto* dq_output = AddQDQNodePair<InputType>(builder, input_arg, .04f,
+                                                (input_min_value + input_max_value) / 2 + 1);
+    builder.AddNode("Conv", {dq_output, dq_w_output, dq_bias_output}, {conv_output});
+
+    auto* q_output = builder.MakeIntermediate();
+    builder.AddQuantizeLinearNode<OutputType>(conv_output, .039f,
+                                              (OutputLimits::min() + OutputLimits::max()) / 2 + 1,
+                                              q_output);
+
+    builder.AddDequantizeLinearNode<OutputType>(q_output, .039f,
+                                                (OutputLimits::min() + OutputLimits::max()) / 2 + 1,
+                                                output_arg);
+  };
+}
+
+template <typename InputType, typename WeightType, typename BiasType, typename OutputType>
+CheckTestGraphFn CheckTestGraph() {
```

Comment:
Not sure if we want to extract the check_test_graph func out too, but I have moved it out for now

Reply:
We may not need this in a helper function, the verification from QDQ transformer tests and NNAPI tests will be very different